pkg/validator/output.go

```diff
@@ -67,15 +75,43 @@ type Result struct {
 	CreatedTime time.Time
 }
 
+func (res *Result) removeSuccessfulResults() {
+	for k, resultMessage := range res.Results {
+		if resultMessage.Success {
```

Comment:
I think we can define this logic once for `ResultSet`, and then call `res.Results.removeSuccessfulResults()` here and below

i.e.
```go
func (rs *ResultSet) {
  for k, resultMessage := range res.Results {
    if resultMessage.Success {
      delete(rs, k)
    }
  }
}
```